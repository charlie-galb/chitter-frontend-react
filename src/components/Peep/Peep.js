import React, { useContext, useState } from 'react';
import { Card, Button } from 'react-bootstrap';
import axios from 'axios';
import { UserContext } from '../../contexts/UserContext.js';

const Peep = (props) => {

    const { userId, currentSessionKey, userHandle } = useContext(UserContext) 
    const likesUserIdArray = props.peepData.likes.map(like => like.user.id)

    const handleLike = async (event) => {
        event.preventDefault();
        try {
            console.log(currentSessionKey)
            const response = await axios.put(`https://chitter-backend-api-v2.herokuapp.com/peeps/${props.peepData.id}/likes/${userId}`, 
            {params: {}}, 
            {headers: {
              Authorization: `Token ${currentSessionKey}` 
            }});
            if (response.data.user.id === userId) {
                console.log(response.data)
                props.retrievePeeps()
            }
        } catch (error) {
            console.log("Error:", error)
        }
    }

    const handleUnlike = async (event) => {
        event.preventDefault();
        try {
            console.log(currentSessionKey)
            const response = await axios.delete(`https://chitter-backend-api-v2.herokuapp.com/peeps/${props.peepData.id}/likes/${userId}`,  
            {headers: {
              Authorization: `Token ${currentSessionKey}` 
            }});
            props.retrievePeeps()
        } catch (error) {
            console.log("Error:", error)
        }
    }

    const handleDelete = async (event) => {
        event.preventDefault();
        try {
            await axios.delete(`https://chitter-backend-api-v2.herokuapp.com/peeps/${props.peepData.id}`, 
            {headers: {
                  Authorization: `Token ${currentSessionKey}` 
                }});
            props.retrievePeeps()
        } catch (error) {
            console.log("Error:", error)
        }
    }

    const renderDeleteButton = () => {
        if (userId == props.peepData.user.id) {
            return <Button variant="secondary" size="sm" onClick={handleDelete} data-testid='delete-button'>Delete</Button>
        }
    }

    const renderLikeButton = () => {
        if ((userId != props.peepData.user.id) && (!likesUserIdArray.includes(userId))) {
            return <Button variant="secondary" size="sm" onClick={handleLike} data-testid='like-button'>Like</Button>
        }
    }

    const renderUnlikeButton = () => {
        if ((userId != props.peepData.user.id) && (likesUserIdArray.includes(userId))) {
            return <Button variant="secondary" size="sm" onClick={handleUnlike} data-testid='unlike-button'>Unlike</Button>
        }
    }

    return (
        <div>
            <Card border="dark" style={{ width: '18rem' }}>
                <Card.Header>{props.peepData.user.handle}</Card.Header>
                <Card.Subtitle className="mb-2 text-muted">Posted at {props.peepData.created_at}</Card.Subtitle>
                <Card.Body>
                    <Card.Text>{props.peepData.body}</Card.Text>
                    {renderDeleteButton()}
                    {renderLikeButton()}
                    {renderUnlikeButton()}
                    <Card.Footer className="text-muted" data-testid='like-count'>Liked by {props.peepData.likes.length}</Card.Footer>
                </Card.Body>
            </Card>
            <br />
        </div>
    )
}

export default Peep;