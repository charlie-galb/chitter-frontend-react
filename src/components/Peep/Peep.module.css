.peepBox {
    background-color: white;
    margin-left: 10px;
    margin-right: 10px;
    text-align: left;
    padding-left: 15px;
    padding-bottom: 5px;
    border-bottom: solid rgb(187, 189, 188) 1px;
}

.peepHandle {
    display: inline;
}

.peepFooter {
   display: flex;
   flex-direction: row;
}

.peepTime {
    display: inline;
    color: rgb(119, 118, 118);
}

.icon {
    font-size: 150%;
    color: #00a56b;
    margin-left: 50%;
}