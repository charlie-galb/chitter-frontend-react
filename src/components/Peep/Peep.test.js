import React from 'react';
import { render, screen } from "@testing-library/react";
import Peep from './Peep.js';

 describe("PeepList", () => {

    const mockPeepData = {
        "id": 3,
        "body": "my first peep :)",
        "created_at": "2018-06-23T13:21:23.317Z",
        "updated_at": "2018-06-23T13:21:23.317Z",
        "user": {
          "id": 1,
          "handle": "kay"
        },
        "likes": [{
          "user": {
            "id": 1,
            "handle": "kay"
          }
        }]
    }
  
  test('renders without crashing', () => {
    const peep = render(<Peep key='1' peepData={mockPeepData}/>);
    screen.debug()
    expect(peep).toMatchSnapshot()
  });

 }) 