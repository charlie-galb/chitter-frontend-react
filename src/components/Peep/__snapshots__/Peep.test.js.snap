// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peep renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="peepBox"
        >
          <div
            class="peepHeader"
          >
            <h4
              class="peepHandle"
            >
              kay
            </h4>
            <p
              class="peepTime"
              data-testid="peep-time-stamp"
            >
               - 
              13:21
               on 
              Tue Jan 23 2018
            </p>
          </div>
          <div
            class="peepBody"
          >
            <p>
              my first peep :)
            </p>
          </div>
          <div
            class="peepFooter"
            data-testid="like-count"
          >
            <p>
              <i>
                1 person likes this
              </i>
            </p>
            <div
              data-testid="delete-icon"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                />
              </svg>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="peepBox"
      >
        <div
          class="peepHeader"
        >
          <h4
            class="peepHandle"
          >
            kay
          </h4>
          <p
            class="peepTime"
            data-testid="peep-time-stamp"
          >
             - 
            13:21
             on 
            Tue Jan 23 2018
          </p>
        </div>
        <div
          class="peepBody"
        >
          <p>
            my first peep :)
          </p>
        </div>
        <div
          class="peepFooter"
          data-testid="like-count"
        >
          <p>
            <i>
              1 person likes this
            </i>
          </p>
          <div
            data-testid="delete-icon"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
              />
            </svg>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
